### PR TITLE
Web docs update regarding migration python interpreter from 3.7.10 to 3.8.12

### PR DESCRIPTION
**Pull request by ashish-patwal**
This PR contains changes to the web docs to provide easy shifting from python 3.7.10 to 3.8.12. This PR is in made in accordance to the following [python migartion PR](https://github.com/oppia/oppia/pull/15508)

### CODE REVIEW

File: Upgrading-python-interpreter-to-version-3.8-for-existing-users.md
Comment (vojtechjelinek) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-users.md]: Hmm, I though we will be creating a new folder in order to upgrade to Python 3.8 and if there are no issues we can later tell the developer to remove the old folder with 3.7

**Review by vojtechjelinek — CHANGES_REQUESTED** [review]
On bigger comment.

File: Upgrading-python-interpreter-to-version-3.8-for-existing-users.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-users.md]: yes sure. I'll fix that asap

**Comment by ashish-patwal** [comment]
PTAL @vojtechjelinek @seanlip

**Comment by seanlip** [comment]
Hi @ashish-patwal, if you want someone to take a look, please use the assignees field. Otherwise they won't. Thanks!

File: _Sidebar.md
Comment (seanlip) [inline _Sidebar.md]: Maybe make the page name "Upgrading python version to 3.8.12"

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: I think "Currently" will go out of date quite soon. Maybe change this to something like: "Oppia changed the Python version it uses from 3.7.10 to 3.8.12 in Oct 2022. This page is aimed at developers who are currently using Python 3.7.10 and need to upgrade to 3.8.12."

Then you can have a new paragraph starting "For effective and effortless..." etc.

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: local --> local repo

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: Add colon at end

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: Inorder --> In order

Add a comma after "not"

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: In the case --> If

Add a comma before the bolded part.

Else if you are not a direnv user you --> Otherwise, you

instructions from --> the instructions in Step 2(b) instead.

(Also I think it's confusing for step 2(a) to say "go to step 2(b)". Could you please refactor this so that "Step 2" is "determine whether you're using direnv and go to step 3(a)/3(b) as appropriate, and Step 3(a)/3(b) are the ones which handle direnv vs pyenv?)

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: Add colon after "this"

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: Add colon after "setup"

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: See above re putting this determination into its own separate step and not conflating it with the actual setup step.

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: Call these "Do the following:" or similar to disambiguate from "Steps", which you're already using one level higher. Ditto elsewhere.

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: ... we will set up ...

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: setup --> set up

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: right --> a

follow the steps mentioned below --> do the following 

(and remove the "Steps" header below).

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: Explain what they should see?

**Review by seanlip — CHANGES_REQUESTED** [review]
Thanks @ashish-patwal! Took a pass.

File: _Sidebar.md
Comment (ashish-patwal) [reply _Sidebar.md]: done

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: done

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: done

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: done

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: done

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: done

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: done

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: done

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: done

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: done

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: done

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: done

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: it's --> its

Actually this seems to introduce a bunch of grammar mistakes. I thought I provided a clean version in my previous comment? (Please read it carefully.)

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: This page is aimed at developers who want to upgrade their Python interpreter in their existing Oppia directory, instead of creating a separate copy of Oppia with the new interpreter.

Before starting, please make sure that your local repo, as well as your remote fork on GitHub, are both up-to-date with the `develop` branch.

(Note: You don't need the bullet points. Just use two paragraphs.)

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: Just to check, do they need to merge develop into all their open/active branches (for which they have a PR out), or is this not mandatory?

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (seanlip) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: What do they do if neither .envrc nor .python-version is present?

**Review by seanlip — CHANGES_REQUESTED** [review]

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: I think it depends. If they are locally working on their PR they would execute the scripts, run test coverage, write lint test etc. In that case the script would throw error since the new installation is done according to python 3.8 , therefore they world need to merge the latest from develop. Otherwise if it doesn't involve testing on local machine, it should not be a compulsion in that case.

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: Either one of them should be present . because that's how the virtualenv managers work. As we enter a directory containing these specific files, the virtualenv managers will activate the environment . And as we leave the directory the environment gets deactivated automatically. So that we don't have to manually switch the version everytime.

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: Done !!

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: Done !!

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (vojtechjelinek) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]:
```suggestion
> **Note**
> Follow this step if you are using direnv as your virtualenv manager.
```

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (vojtechjelinek) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]:
```suggestion
> **Note**
> Follow this step if you are using pyenv as your virtualenv manager.
```

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (vojtechjelinek) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]:
```suggestion
Our first step is to install the new interpreter version `3.8.12` through pyenv by issuing the following command through a terminal:
```

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (vojtechjelinek) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]:
```suggestion

After you execute the above command you can expect some output as depicted below:
```

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (vojtechjelinek) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]:
```suggestion
2. Run this command in the same directory to complete the setup:
```

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (vojtechjelinek) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]:
```suggestion
#### Do the following:
```

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (vojtechjelinek) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]:
```suggestion
    The above command will produce output as depicted below:
```

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (vojtechjelinek) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]:
```suggestion
To verify that you have a working installation, do the following:

1. Verify you are using `python 3.8.12` on entering Oppia directory by running `python --version` in your Oppia directory. You should see the following output:
```

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (vojtechjelinek) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]:
```suggestion

```

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (vojtechjelinek) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]:
```suggestion

1. Edit the `.envrc` file in your oppia folder to this:
```

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (vojtechjelinek) [inline Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: Please do not use \`xxx\` for files so that they are not confused with commands. Use _italics_ for example

**Review by vojtechjelinek — CHANGES_REQUESTED** [review]
Thanks! Took a pass

**Review by seanlip — APPROVED** [review]
Thanks! No further concerns from my end.

File: Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md
Comment (ashish-patwal) [reply Upgrading-python-interpreter-to-version-3.8-for-existing-oppia-directory.md]: Done !!

**Review by vojtechjelinek — APPROVED** [review]
LGTM!

**Review by vojtechjelinek — CHANGES_REQUESTED** [review]
Blocking until we can merge this.

**Comment by vojtechjelinek** [comment]
**DO NOT MERGE**

**Comment by vojtechjelinek** [comment]
READY TO MERGE

**Pull request merged**
